tracer/src/Datadog.Trace.OpenTracing/OpenTracingSpanBuilder.cs

```diff
@@ -70,12 +70,12 @@ public ISpanBuilder IgnoreActiveSpan()
             return this;
         }
 
-        public ISpan Start()
+        public global::OpenTracing.ISpan Start()
         {
             lock (_lock)
             {
                 ISpanContext parentContext = GetParentContext();
-                Span ddSpan = _tracer.DatadogTracer.StartSpan(_operationName, parentContext, _serviceName, _start, _ignoreActiveSpan);
+                ISpan ddSpan = _tracer.DatadogTracer.StartSpan(_operationName, parentContext, _serviceName, _start, _ignoreActiveSpan);
```

Comment:
Everything else in this PR seems fine except for this file. We removed public StartSpan/Activatee API's but we still use them in the OpenTracing codepaths which are publicly accessible here. The good thing is that the Span object is internal, but I'll need to think about if there can be unexpected behavior when using these API's..

Reply:
I added a new interface called `IDatadogOpenTracingTracer` (open to suggestions) that defines the minimum interface needed from the OpenTracing side to interact with `Datadog.Trace.Tracer`, since it is possible on .NET Core for the `Datadog.Trace.dll` to be promoted to a newer version provided by automatic instrumentation while `Datadog.Trace.OpenTracing.dll` remains unchanged. Besides that communication, I think the changes in this PR are fine, because the OpenTracing side of the fence maintains its own ScopeManager and "Active" span